sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -582,6 +609,33 @@ public static <T> T getResultWithTimeout(Future<T> future, Duration timeout)
         return ImplUtils.getResultWithTimeout(future, timeout.toMillis());
     }
 
+    /**
+     * Submits a task for execution on the shared executor service.
+     *
+     * @param <T> The type of value returned by the {@code callable}.
+     * @param callable The task to submit for execution.
+     * @return A {@link Future} representing the pending completion of the task.
+     */
+    public static <T> Future<T> submit(Callable<T> callable) {
```

Comment:
I am not convinced that we want this to hang off of CoreUtils. I wonder whether we have a (warning: placeholder name) `SharedThreadPool` class that centralises this. It would allow for use-case-specific customisation / subclassing, and different thread pools for different use cases.